Adds and improves logging for various shit.

Pickle-Coding opened the pull request:
## About The Pull Request
Adds logging for RCD construction and deconstruction. Hallucinated projectiles no longer causes logs. Flamethrowers log gas mixture information, the flamethrower, the gas tank, tank distribution pressure and whether it was lit. Adds a lot more logging to records consoles. Frozen objects now log when they get shattered. 
## Why It's Good For The Game
Closes #68452 
Closes #71798 
Closes #78008 
Closes #81098 
Closes #81130 
## Changelog
:cl: Pickle-Coding and Rhials
admin: RCD construction and deconstruction are logged.
admin: Hallucinated projectiles no longer log.
admin: Gives more detail to flamethrower logging.
admin: More actions are logged for records consoles usage.
admin: Frozen object shattering is logged.
/:cl:

Rhials committed:
freeze

Rhials committed:
whoops

Rhials committed:
NOT DONE YET I JUST FORGOT TO PUSH CHANGES

Rhials committed:
Merge branch 'master' into freeze-logs-legion

Pickle-Coding committed:
Logs RCD usage.

rcd_create() gets logged.

Pickle-Coding committed:
Hallucinated projectiles no longer log.

Pickle-Coding committed:
Flamethrower improved logging.

Flamethrower logging now includes the gas mixture, flamethrower, igniter, tank and lit status. Also logs when attempted to use with pacifism.

Pickle-Coding committed:
Just to be safe.

Pickle-Coding committed:
Merge branch 'freeze-logs-legion' of https://github.com/Rhials/tgstation into logging

Pickle-Coding committed:
Frozen object shatter logging.

Pickle-Coding committed:
Records console logging.

Improves and add more logging for records console and medical console records modification.

Pickle-Coding committed:
Better flamethrower logging.

Flamethrower logging now includes the tank distribution pressure. It also wraps the values with quotations or braces.

SyncIt21 commented on code/game/objects/items/rcd/RCD.dm:
```diff
@@ -221,6 +221,7 @@
 	current_active_effects += 1
 	_rcd_create_effect(target, user, delay, rcd_results)
 	current_active_effects -= 1
+	log_tool("used RCD with design mode: \"[rcd_results["[RCD_DESIGN_MODE]"]]\" and design path: \"[rcd_results["[RCD_DESIGN_PATH]"]]\" with delay [delay]", user)
```
Makes sense to log only if the effect was successful.

Luckily _rcd_create_effect() proc returns TRUE if it was so you can write an if condition there and log only if that proc returns TRUE.

Pickle-Coding committed:
Only log RCD if successful.

Prevents misleading logs.

ZephyrTFA commented on code/__HELPERS/logging/atmos.dm:
```diff
@@ -1,8 +1,5 @@
 /// Logs the contents of the gasmix to the game log, prefixed by text
```
make this use the data list

Pickle-Coding replied:
What is the data list used for?

SyncIt21 replied:
The data list is your user defined map of key(string) to value(can be anything  but is converted to string anyway) which is written in json format to your log file.

The name of your file where this data is stored is <our usual log file>-json.json.  obviously json format can be parsed later for statictics and what not

In your case vars like temperature and moles can be put in list format as 2 seperate keys and passed to log game.

Message should only be used to explain what vars are being logged and not store the actual data itself

Pickle-Coding replied:
Done.

Pickle-Coding committed:
log_atmos() json data.

log_atmos() passes information of the mixture and gases into the log_game() data. Also fixes gas tank release not logging properly.

Pickle-Coding committed:
Whoops.

Scales the value for logging with the gas moles.

Pickle-Coding commented:
Drafting because there are a few issues with my PR that I'll fix later.

Pickle-Coding committed:
More useful RCD logging + atmos log fix.

Changes RCD logging so admins don't have to codedive to find the bitflag defines to understand the results.

/datum/gas_mixture's remove() and remove_ratio() proc will now always pass in the volume for creating the removed gas mixture. This corrects the logging for gas tank release so the reported volume is correct.

Adds more information for log_atmos().

Pickle-Coding committed:
Merge branch 'master' into logging

ZephyrTFA commented:
go away stalebot
